lib/submit-request.js

```diff
@@ -185,6 +185,39 @@ SubmitRequest.prototype.commit = function(callback) {
   });
 };
 
+SubmitRequest.prototype._metadataProjection = function() {
+  var request = this;
+
+  // Default behavior
+  if (!request.opMetadataProjection) {
+    return undefined;
+  }
+
+  // Granular projection
+  if (typeof request.opMetadataProjection === 'object') {
+    return request._granularMetadataProjection();
+  }
+
+  // Full projection
+  return request.op.m;
+};
+
+SubmitRequest.prototype._granularMetadataProjection = function() {
+  var request = this;
+  var metadataProjection = {};
+  for (var key in request.opMetadataProjection) {
```

Comment:
We should at least add a code comment here that we only support a single level of projection.

Would be extra nice if we updated the docs to include this flag and notes on its use.

Reply:
Added a comment, and updated [docs](https://github.com/share/sharedb/pull/586/files#diff-e185cf468e72547cbd5b8e2dd6f5604be3330207c759de532c540c9a2702cf5d) with an example.